src/modules/staking/staking.service.ts

```diff
@@ -80,12 +78,9 @@ export class StakingService implements OnModuleDestroy {
     });
   }
   private InitEventListeners(): void {
-    this.stakingPoolFactory.addListener(
-      'StakingPoolLaunched',
-      async (org, address) => {
-        await this.saveStakingPool(address);
-      },
-    );
+    this.stakingPoolFactory.on('StakingPoolLaunched', async (_, address) => {
```

Comment:
curious question here:  is `on` equivalent to `addListener`  in the sense that it initialises and adds the listener? if not,  just thinking that we might no longer need the removeListener part.

Reply:
There is no longer `addListener` method, just `on` and under the hood it is calling `_addEventListener`. So I think we still need to `removeListener` part.